controllers/idler/idler_controller.go

```diff
@@ -460,196 +464,90 @@ func getSupportedScaleResource(ownerReference metav1.OwnerReference) *schema.Gro
-			return owner.Kind, owner.Name, true, nil
+			return nil
 		}
-		return "", "", false, err
+		return err
 	}
-	log.FromContext(ctx).Info("DaemonSet deleted", "name", ds.Name)
-	return owner.Kind, owner.Name, true, nil
+	log.FromContext(ctx).Info("DaemonSet deleted", "name", object.GetName())
+	return nil
 }
 
-func (r *Reconciler) scaleStatefulSetToZero(ctx context.Context, namespace string, owner metav1.OwnerReference) (string, string, bool, error) {
-	s := &appsv1.StatefulSet{}
-	if err := r.AllNamespacesClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: owner.Name}, s); err != nil {
-		if apierrors.IsNotFound(err) { // Ignore not found errors. Can happen if the parent controller has been deleted. The Garbage Collector should delete the pods shortly.
-			return owner.Kind, owner.Name, true, nil
-		}
-		return "", "", false, err
-	}
-	zero := int32(0)
-	patched := s.DeepCopy()
-	patched.Spec.Replicas = &zero
-	if err := r.AllNamespacesClient.Patch(ctx, patched, client.MergeFrom(s)); err != nil {
-		return "", "", false, err
+func (r *Reconciler) scaleStatefulSetToZero(ctx context.Context, object client.Object) error {
+	patch := []byte(`{"spec":{"replicas":0}}`)
+	if err := r.AllNamespacesClient.Patch(ctx, object, client.RawPatch(types.MergePatchType, patch)); err != nil {
+		return err
 	}
-	log.FromContext(ctx).Info("StatefulSet scaled to zero", "name", s.Name)
-	return owner.Kind, owner.Name, true, nil
+	log.FromContext(ctx).Info("StatefulSet scaled to zero", "name", object.GetName())
+	return nil
 }
 
-func (r *Reconciler) scaleDeploymentConfigToZero(ctx context.Context, namespace string, owner metav1.OwnerReference) (string, string, bool, error) {
-	dc := &openshiftappsv1.DeploymentConfig{}
-	if err := r.AllNamespacesClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: owner.Name}, dc); err != nil {
-		if apierrors.IsNotFound(err) { // Ignore not found errors. Can happen if the parent controller has been deleted. The Garbage Collector should delete the pods shortly.
-			return owner.Kind, owner.Name, true, nil
-		}
-		return "", "", false, err
-	}
-	patched := dc.DeepCopy()
-	patched.Spec.Replicas = 0
-	patched.Spec.Paused = false
-	if err := r.AllNamespacesClient.Patch(ctx, patched, client.MergeFrom(dc)); err != nil {
-		return "", "", false, err
+func (r *Reconciler) scaleDeploymentConfigToZero(ctx context.Context, object client.Object) error {
+	patch := []byte(`{"spec":{"replicas":0,"paused":false}}`)
+	if err := r.AllNamespacesClient.Patch(ctx, object, client.RawPatch(types.MergePatchType, patch)); err != nil {
+		return err
 	}
-	log.FromContext(ctx).Info("DeploymentConfig scaled to zero", "name", dc.Name)
-	return owner.Kind, owner.Name, true, nil
+	log.FromContext(ctx).Info("DeploymentConfig scaled to zero", "name", object.GetName())
+	return nil
```

Comment:
I guess we can replace most of those `scaleXtoZero()` methods by a generic `scaleControllerReplicasToZero()`  or something like this, right? The only difference between them is just logging if I'm not missing anything.

Reply:
Yes, this is exactly the duplication I was talking about. Let me do the change in smaller PRs. For this case, in order to minimize the potential damage, I wanted to "only" introduce the owner fetcher, but still keep the logic similar to the previous format.